clarity/src/vm/contexts.rs

```diff
@@ -1263,20 +1267,36 @@ impl<'a, 'b, 'hooks> Environment<'a, 'b, 'hooks> {
         contract_content: &str,
         ast_rules: ASTRules,
     ) -> Result<()> {
+        use super::database::MemoryBackingStore;
+
         let clarity_version = self.contract_context.clarity_version.clone();
 
-        let contract_ast = ast::build_ast_with_rules(
+        let mut contract_ast = ast::build_ast_with_rules(
             &contract_identifier,
             contract_content,
             self,
             clarity_version,
             self.global_context.epoch_id,
             ast_rules,
         )?;
+
+        let mut store = MemoryBackingStore::new();
```

Comment:
Is the idea that this will go to a real store later on? Feels like a pretty heavy operation to just throw away in a memory store?

Reply:
This should actually be using the `contract_analysis` that is already done. I forgot I put this here to test it out before changing the various function signatures to pass that one in.

Reply:
Oh, no... I didn't forget about this, I just forgot about why! This code is not reached during normal execution. This path is only hit from tests and from the clarity CLI, so it's fine. I'll add a comment.

Reply:
Added in f945ea56d.